sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/BaseAuthorizationTokenProvider.java

```diff
@@ -49,6 +49,7 @@ private static String getResourceSegment(ResourceType resourceType) {
         case Document:
             return Paths.DOCUMENTS_PATH_SEGMENT;
         case DocumentCollection:
+        case PartitionKey:
```

Comment:
You might need to added support for resource token requests as it is introducing new resource type , please check method https://github.com/Azure/azure-sdk-for-java/blob/c28d9a2e4b40feece2a26799d0fecb31d6721b84/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/BaseAuthorizationTokenProvider.java#L177
to verify you can add a simple test in this test class https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ResourceTokenTestForV4.java